text/trace/0173-messaging-semantic-conventions.md

```diff
@@ -0,0 +1,242 @@
+
+### Producer
+
+The "producer" is a specific instance, process or device that creates and
+publishes a message. "Publishing" is the process of sending a message or batch
+to the intermediary or consumer.
+
+### Consumer
+
+A "consumer" receives the message and acts upon it. It uses the context and
+data to execute some logic, which might lead to the occurrence of new events.
+
+The consumer receives, processes, and settles a message. "Receiving" is the
+process of obtaining a message from the intermediary, "processing" is the
+process of acting on the information a message contains, "settling" is the
+process of notifying an intermediary that a message was processed successfully.
+
+### Intermediary
+
+An "intermediary" receives a message to forward it to the next receiver, which
+might be another intermediary or a consumer.
+
+## Scenarios
+
+Producing and consuming a message involves five stages:
+
+```
+PRODUCER
+
+Create
+  |                            CONSUMER
+  v        +--------------+                   
+Publish -> | INTERMEDIARY | -> Receive
+           +--------------+       |
+                  ^               v
+                  .            Process
+                  .               |
+                  .               v
+                  . . . . . .  Settle
+```
+
+1. The producer creates a message.
+2. The producer publishes the message to an intermediary.
+3. The consumer receives the message from an intermediary.
+4. The consumer processes the message.
+5. The consumer settles the message by notifying the intermediary that the
+   message was processed. In exceptional cases (fire-and-forget), the
+   settlement stage does not exist.
+
+The messaging semantic conventions need to define how to model those stages in
+traces, how to propagate context, and how to enrich traces with attributes.
+Failures and retries need to be handled in all stages that interface with the
+intermediary (publish, receive and settle) and will be covered by general
+instrumentation guidance.
+
+Based on this model, the following scenarios capture major requirements and
+can be used for prototyping, as examples, and as test cases.
+
+### Individual settlement
+
+Individual settlement systems imply independent logical message flows. A single
+message is created and published in the same context, and it's delivered,
+consumed, and settled as a single entity. Each message needs to be settled
+individually. Usually, settlement information is stored by the intermediary, not
+by the consumer.
+
+Transport batching can be treated as a special case: messages can be
+transported together as an optimization, but are produced and consumed
+individually.
+
+```
++---------+ +---------+ +---------+ +---------+ +---------+ +---------+
+|Message A| |Message B| |Message C| |Message D| |Message E| |Message F|
++---------+ +---------+ +---------+ +---------+ +---------+ +---------+
+ Settled                 Settled                             Settled 
```

Comment:
This figure is not clear to me. Is there an x-axis that's supposed to present time or does this just show a mapping of `message`->`settled: yes/no` and could just as well be a table?

Reply:
I added a textual description.